File: app_canibuythat/src/main/java/com/gb/canibuythat/di/CanIBuyThatComponent.java
package com.gb.canibuythat.di;

import android.app.Application;

import javax.inject.Singleton;

import dagger.BindsInstance;
import dagger.Component;

@Singleton
@Component(modules = {CanIBuyThatModule.class})
public interface CanIBuyThatComponent extends CanIBuyThatGraph {

    @Component.Builder
    interface Builder {
        @BindsInstance
        Builder application(Application application);

        CanIBuyThatComponent build();
    }
}

File: app_canibuythat/src/main/java/com/gb/canibuythat/di/CanIBuyThatComponent.kt
package com.gb.canibuythat.di

import android.app.Application
import dagger.BindsInstance
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = arrayOf(CanIBuyThatModule::class))
interface CanIBuyThatComponent : CanIBuyThatGraph {

    @Component.Builder
    interface Builder {
        @BindsInstance
        fun application(application: Application): Builder

        fun build(): CanIBuyThatComponent
    }
}
